bls/bls-core/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "bls-core"
+edition = { workspace = true }
+
+[dependencies]
+blst = { workspace = true }
```

Comment:
We need to get rid of `blst` dependency here. Otherwise we still need to compile `blst` if we don't use `blst` as a backend.